### ISSUE TITLE
Return default value on failed lookup

### ISSUE THREAD

**Issue by pote** [tonchis/fat#4]
I'd like Fat to return a default value on failed lookups - basically emulating plain `Hash#fetch` usage.

Example:

```ruby
Fat.at({}, :one, :two, :three, default: 'caramba!') #=> 'caramba!'
```

**Comment by tonchis** [comment]
@pote while looking into this I noticed that the above is not permitted Ruby syntax. Keyword arguments and argument list can't be used at the same time:

```ruby
def foo(default: 'caramba', *fields)
# SyntaxError: unexpected *
# def foo(default: 'caramba', *fields)
#                              ^
```

The alternative I'm thinking about is turning the argument list into an array:

```ruby
Fat.at({}, [:one, :two, :three], default: "caramba!")
```

What do you think?

**Comment by tonchis** [comment]
@pote Actually, your example is correct! The argument list just needs to be before the named argument.

```ruby
def foo(hash, *fields, default: 'caramba')
end
```

It can be done :)

**Comment by tonchis** [comment]
This is implemented in v2.0.1; [test here](https://github.com/tonchis/fat/blob/dea15312cd43885db662919ec04b76d4cb8d6bdd/test/fat_test.rb#L27-L30)

**Comment by pote** [comment]
Thanks!!!!! :heart: :dancer: :tada:
